File: test/fakers/checkout/cart_item_faker.dart
import 'package:faker/faker.dart';
import 'package:sertton/core/checkout/dtos/cart_item_dto.dart';

class CartItemFaker {
  static final _faker = Faker();

  static CartItemDto fakeDto({
    String? productId,
    String? skuId,
    int? quantity,
  }) {
    return CartItemDto(
      productId: productId ?? _faker.guid.guid(),
      skuId: skuId ?? _faker.guid.guid(),
      quantity: quantity ?? _faker.randomGenerator.integer(10, min: 1),
    );
  }

  static List<CartItemDto> fakeManyDto({
    int count = 10,
    String? productId,
    String? skuId,
    int? quantity,
  }) {
    return List.generate(
      count,
      (index) =>
          fakeDto(productId: productId, skuId: skuId, quantity: quantity),
    );
  }
}

File: src/core/checkout/entities/fakers/cart-item-faker.ts
import { faker } from "@faker-js/faker"

import { ProductFaker } from "@/core/catalog/entities/fakers/product-faker"

import type { CartItem } from "../cart-item"

export class CartItemFaker {
  static fake(cartItem?: Partial<CartItem>): CartItem {
    const product = ProductFaker.fake()

    return {
      product,
      productId: product.id,
      quantity: faker.number.int({ min: 1, max: 5 }),
      skuId: faker.string.uuid(),
      ...cartItem,
    }
  }

  static fakeMany(count = 10, cartItem?: Partial<CartItem>): CartItem[] {
    return Array.from({ length: count }, () => CartItemFaker.fake(cartItem))
  }
}
